src/sessions.py

```diff
@@ -0,0 +1,142 @@
+# -*- coding: utf-8 -*-
+
+#   Copyright (c) 2010-2014, MIT Probabilistic Computing Project
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from __future__ import print_function
+import bayeslite
+import time
+import sys
+import json
+import requests
+
+class SessionTracer(object):
+
+    def __init__(self, bdb):
```

Comment:
optionally pass in a logging object so we can send messages e.g. to ipython?  Rather than calling it _print, I'd call .info(msg), .warn(msg), .error(msg) on that instead. The default if it's null can be to print all of those.